prime-router/docs/proposals/0017-metadata-tables-cleanup.md

```diff
@@ -0,0 +1,441 @@
+comment on column state.country_id is
+    'The foreign key for the country the state is in';
+
+
+-- a county table
+create table county
+(
+    county_id                   bigserial not null primary key
+    , county_name               varchar(512) not null
+    , county_fips_code          varchar(50) null
+    , state_id                  bigint not null references state(state_id)
+);
+
+comment on table county is
+    'A table for the county an address can be in.';
+comment on column county.county_id is
+    'The primary key for the county table';
+comment on column county.county_name is
+    'The name of the county';
+comment on column county.county_fips_code is
+    'The numeric code for the county per FIPS';
+comment on column county.state_id is
+    'The foreign key for the state a county is in. Per the Census Bureau no county can cross state lines';
+
+
+-- a facility could exist in multiple roles: ordering facility, testing facility,
+-- but in the future it could also be a servicing facility for vaccines or patient care
+-- at a hospital
+create table facility_role
+(
+    facility_role_id                    bigserial not null primary key
+    , facility_role                     varchar(512) not null
+);
+
+comment on table facility_role is
+    'The different kinds of roles a facility can play, for example an ordering facility, a reporting facility, etc';
+comment on column facility_role.facility_role_id is
+    'The primary key for the facility roles';
+comment on column facility_role.facility_role is
+    'The type of role a facility can play';
+
+-- insert the first few rows
+insert into facility_role (facility_role) values ('ordering');
+insert into facility_role (facility_role) values ('testing');
+insert into facility_role (facility_role) values ('reporting');
+
+
+-- a unified list of facilities
+-- TODO: should we separate out the address from the facility? can a facility have more than one address?
+create table facility
+(
+    facility_id                         bigserial not null primary key
+    , facility_name                     varchar(512) not null
+    , facility_clia                     varchar(50) null
+    , facility_oid                      varchar(512) null
+    , facility_namespace_id             varchar(512) null
+    , facility_street                   varchar(512) null
+    , facility_street2                  varchar(512) null
+    , facility_city                     varchar(128) null
+    , facility_postal_code              varchar(50) null
+    , facility_callback_number          varchar(512) null
+    , facility_county_id                bigint null references county(county_id)
+    , facility_state_id                 bigint null references state(state_id)
+    , facility_country_id               bigint null references country(country_id)
+    , has_valid_clia                    boolean not null default true
```

Comment:
how will this be determined? What will be the source of truth for if something is a valid CLIA?. Will we use the existing regex that checks the format or some other source?

Reply:
@GilmoreA6 - Good question. I think it will be a combination of doing actual lookups at some point on the CLIAs, as well as us setting the values manually for some testing facilities.

Reply:
towards this question: can a facility have more than one address?

ORC-22 is [repeatable](https://hl7-definition.caristix.com/v2/HL7v2.5.1/Fields/ORC.22)

A tricky part is that [ORC-21](https://hl7-definition.caristix.com/v2/HL7v2.5.1/Fields/ORC.21) and [ORC-22](https://hl7-definition.caristix.com/v2/HL7v2.5.1/Fields/ORC.22) are also repeatable. But I don't know if we can rely on senders to ensure that Ordering Facility Name [1] corresponds to Ordering Facility Address [1] and Ordering Facility Phone [1] then Ordering Facility Name [2] corresponds to Ordering Facility Address [2] and Ordering Facility Phone [2]

I think having repeats in these fields is an edge case. More likely is that a facility will change addresses or sometimes the address will be of the specific lab building then other times it will be the general address for the entire facility. The same could be said of the phone number